### ISSUE TITLE
Color image training for infrared camera detection

### ISSUE THREAD

**Issue by 1623021453** [ultralytics/ultralytics#21465]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

When I train the YOLO11 detection model with color images, the effect is not good when using an infrared camera for detection. Will the detection effect of the infrared camera be better if I convert the color images to grayscale images and train the model again? Or can parameters be modified during training to directly train into grayscale images?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @1623021453, thank you for reaching out and for your interest in Ultralytics 🚀! For detailed insights and common questions, we recommend checking out the [Docs](https://docs.ultralytics.com/), where you'll find comprehensive [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples.

Since your question relates to data modality and custom training, please provide as much detail as possible, such as sample images from both your color and infrared datasets, your training logs, and any relevant configuration details. This helps us and the community give you more precise guidance. Also, be sure to review our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) to optimize your experiments.

If you believe you've encountered a bug, please supply a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) so we can assist you efficiently 🐛.

You’re also welcome to join the Ultralytics community:
- For real-time chat, visit [Discord](https://discord.com/invite/ultralytics) 🎧
- For in-depth discussions, check [Discourse](https://community.ultralytics.com/)
- To share and discuss with others, try our [Subreddit](https://www.reddit.com/r/Ultralytics/)

## Upgrade

Please ensure you are running the latest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in these up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

This is an automated response 🪄. An Ultralytics engineer will review your question and assist you soon!

**Comment by glenn-jocher** [comment]
Yes, converting to grayscale during training could help bridge the domain gap between color training images and infrared inference. Rather than manually converting all images, you can use [Albumentations integration with YOLO11](https://docs.ultralytics.com/integrations/albumentations/) to apply grayscale transformations during training with the ToGray augmentation - this allows your model to learn from both color and grayscale representations. Ideally though, including some infrared images in your training dataset would provide the best results for infrared camera detection.

**Comment by 1623021453** [comment]
Thank You. Can you explain in detail where to set ToGray and how to set which parameters? I looked at the link you provided, but I didn't understand where to set the parameters.

**Comment by glenn-jocher** [comment]
The ToGray transformation is already built into YOLO11's Albumentations integration and applies automatically when you train - no additional setup needed. It's configured with `p=0.01` (1% probability), `num_output_channels=3` (maintains 3-channel output), and uses the `"weighted_average"` method for natural-looking grayscale conversion. If you want to increase the grayscale probability beyond 1%, you'd need to create a custom augmentation pipeline, but the current setup should help your model learn from both color and grayscale representations during training.

**Comment by 1623021453** [comment]
I modified augment.py：T = [
                A.Blur(p=0.01),
                A.MedianBlur(p=0.01),
                A.ToGray(p=0.9),  #0.01
                A.CLAHE(p=0.01),
                A.RandomBrightnessContrast(p=0.0),
                A.RandomGamma(p=0.0),
                A.ImageCompression(quality_range=(75, 100), p=0.0),
            ]  ,And then train again, is that okay？

**Comment by glenn-jocher** [comment]
Yes, that modification will work perfectly! Changing `A.ToGray(p=0.9)` in the augment.py file will give you a 90% probability of grayscale conversion during training, which should help your model better adapt to infrared camera detection. Just note that you're modifying the source code directly, so if you update Ultralytics in the future, you'll need to reapply this change.
